package.json

```diff
@@ -61,6 +61,13 @@
     "integration/*",
     "repo-scripts/*"
   ],
+  "peerDependencies": {
+    "typescript": ">=4.7.0"
+  },
+  "peerDependenciesMeta": {
+    "typescript": {
+      "optional": true
+    },
```

Comment:
What does a JavaScript user see when they don't meet this optional peer dependency?